docs/source/developers/python.rst

```diff
@@ -390,60 +401,132 @@ First, starting from fresh clones of Apache Arrow:
+- add the path of its built .dll-libraris to ``PATH``
+- and choose the compiler to be used
 
 .. code-block:: shell
 
-   mkdir cpp\build
-   cd cpp\build
-   set ARROW_HOME=C:\thirdparty
-   cmake -G "Visual Studio 14 2015 Win64" ^
-         -DCMAKE_INSTALL_PREFIX=%ARROW_HOME% ^
-         -DARROW_CXXFLAGS="/WX /MP" ^
-         -DARROW_GANDIVA=on ^
-         -DARROW_PARQUET=on ^
-         -DARROW_PYTHON=on ..
-   cmake --build . --target INSTALL --config Release
-   cd ..\..
+   set ARROW_HOME=%cd%\arrow-dist
+   set PATH=%ARROW_HOME%\bin;%PATH%
+   set PYARROW_CMAKE_GENERATOR=Visual Studio 15 2017 Win64
 
-After that, we must put the install directory's bin path in our ``%PATH%``:
+This assumes Visual Studio 2017 or its build tools are used. For Visual Studio
+2015 and its build tools use the following instead:
 
 .. code-block:: shell
 
-   set PATH=%ARROW_HOME%\bin;%PATH%
+   set PYARROW_CMAKE_GENERATOR=Visual Studio 14 2015 Win64
+
+Let's configure, build and install the Arrow C++ libraries:
+
+.. code-block:: shell
+
+   mkdir arrow\cpp\build
+   pushd arrow\cpp\build
+   cmake -G "%PYARROW_CMAKE_GENERATOR%" ^
+       -DCMAKE_INSTALL_PREFIX=%ARROW_HOME% ^
+       -DARROW_CXXFLAGS="/WX /MP" ^
+       -DARROW_GANDIVA=on ^
+       -DARROW_PARQUET=on ^
+       -DARROW_PYTHON=on ^
+       ..
+   cmake --build . --target INSTALL --config Release
+   popd
 
 Now, we can build pyarrow:
 
 .. code-block:: shell
 
-   cd python
+   pushd arrow\python
    set PYARROW_WITH_GANDIVA=1
    set PYARROW_WITH_PARQUET=1
    python setup.py build_ext --inplace
+   popd
+
+.. note::
+
+   For building pyarrow, the above defined environment variables need to also
+   be set. Remember this if to want to re-build ``pyarrow`` after your initial build.
 
 Then run the unit tests with:
 
 .. code-block:: shell
 
+   pushd arrow\python
    py.test pyarrow -v
+   popd
+
+.. note::
+
+   With the above instructions the Arrow C++ libraries are not bundled with
+   the python extension. This is recommended for development as it allows the
+   C++ libraries to be re-built separately.
+
+   As a consequence however, ``python setup.py install`` will also not install
+   the Arrow C++ libraries. Therefore, to use ``pyarrow`` in python, ``PATH`` 
+   must contain the directory with the Arrow .dll-files.
+
+   If you want to bundle the Arrow C++ libraries with ``pyarrow`` add 
+   ``--bundle-arrow-cpp`` as build parameter:
+
+   ``python setup.py build_ext --inplace --bundle-arrow-cpp``
 
 Running C++ unit tests for Python integration
 ---------------------------------------------
 
-Getting ``python-test.exe`` to run is a bit tricky because your
-``%PYTHONHOME%`` must be configured to point to the active conda environment:
+Running C++ unit tests should not be necessary for most developers. If you do
+want to run them, you need to pass ``-DARROW_BUILD_TESTS=ON`` during 
+configuration of the Arrow C++ library build:
+
+.. code-block:: shell
+
+   pushd arrow\cpp\build
```

Comment:
nit: Is it better to add `mkdir arrow\cpp\build`?

Reply:
I think you are quite right. Just in case somebody jumps directly to this section.